.github/workflows/publish.yml

```diff
@@ -0,0 +1,21 @@
+name: Publish (Testing)
```

Comment:
are there best practices on naming conventions for tasks? I'm thinking it's often desirable to refer to them by name, and including things like whitespace and parentheses make it hard to reference.

Reply:
This is just called "Publish" in #247, the "(Testing)" here is so I know that it's this test PR triggering the action, not that one. There are a few things in this action that are only here for debugging, and that shouldn't go into prod. E.g. that we run this in response to push and pull request events, not just releases.

I've been force-pushing the head commit on this branch to test the action, the code in this PR is very much a moving target.